gaitmap_mad/gaitmap_mad/event_detection/_rampp_event_detection.py

```diff
@@ -238,7 +288,160 @@ def _find_all_events(
     )
 
 
-def _detect_ic(
+def _find_all_events_for_ic_stride(
+    gyr: pd.DataFrame,
+    acc: pd.DataFrame,
+    stride_list: pd.DataFrame,
+    events: Tuple[str, ...],
+    ic_search_region: Tuple[float, float],
+    min_vel_search_win_size: int,
+    sampling_rate_hz: float,
+    gyr_ic_lowpass_filter: Optional[BaseFilter],
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """Find events in provided data by looping over single strides."""
+    gyr_ml = gyr["gyr_ml"]
+    if "ic" in events:
+        if gyr_ic_lowpass_filter is not None:
+            gyr_ml_filtered = gyr_ic_lowpass_filter.filter(
+                gyr_ml, sampling_rate_hz=sampling_rate_hz
+            ).filtered_data_.to_numpy()
+        else:
+            gyr_ml_filtered = gyr_ml.to_numpy()
+    else:
+        gyr_ml_filtered = None
+    gyr = gyr.to_numpy()
+    acc_pa = -acc["acc_pa"].to_numpy()  # have to invert acc data to work on rampp paper
+    ic_events = []
+    tc_events = []
+    min_vel_events = []
+    _is_initial_stride = False
+    for index, stride in stride_list.iterrows():
+        current_stride_start = stride["start"]
+        current_stride_end = stride["end"]
+        if "ic" in events:
+            if index == 0:
```

Comment:
This might be dangerous. We don't enforce any specific namings for the index. So it could be that the index is called differently. I would suggest to use `enumerate` to get the first iteration of the loop